packages/server/src/appMigrations/migrations/20240604153647_initial_sqs.ts

```diff
@@ -0,0 +1,51 @@
+import { context } from "@budibase/backend-core"
+import { allLinkDocs } from "../../db/utils"
+import LinkDocumentImpl from "../../db/linkedRows/LinkDocument"
+import sdk from "../../sdk"
+import env from "../../environment"
+import { DBError } from "@budibase/types"
+
+const migration = async () => {
+  const linkDocs = await allLinkDocs()
+
+  const docsToUpdate = []
+  for (const linkDoc of linkDocs) {
+    if (linkDoc.tableId) {
+      // It already had the required data
+      continue
+    }
+
+    // it already has the junction table ID - no need to migrate
+    if (!linkDoc.tableId) {
+      linkDoc.tableId = new LinkDocumentImpl(
+        linkDoc.doc1.tableId,
+        linkDoc.doc1.fieldName,
+        linkDoc.doc1.rowId,
+        linkDoc.doc2.tableId,
+        linkDoc.doc2.fieldName,
+        linkDoc.doc2.rowId
+      ).tableId
+      docsToUpdate.push(linkDoc)
+    }
+  }
+
+  const db = context.getAppDB()
+  if (docsToUpdate.length) {
+    await db.bulkDocs(docsToUpdate)
+  }
+
+  // at the end make sure design doc is ready
+  await sdk.tables.sqs.syncDefinition()
+  // only do initial search if environment is using SQS already
+  // initial search makes sure that all the indexes have been created
+  // and are ready to use, avoiding any initial waits for large tables
+  if (env.SQS_SEARCH_ENABLE) {
```

Comment:
I don't feel too comfortable with it... Should the migration run only if the SQS is enabled? What would happen if it runs when it's not enabled, and then it gets enabled?

Reply:
that's fine - disabling/enabling this stuff is ok, i've tested disabling/enabling over and over again and it doesn't cause any problems (as the migration is backwards compatible and the integrity test confirms you can re-run).

Reply:
The worst case scenario is that the indexes aren't generated by the migration (SQS isn't available, so it just doesn't call it) but these indexes will be generated on the fly as well - essentially the user will have a slow first query experience, but after that problem is solved.